docs/limitations.md

```diff
@@ -0,0 +1,69 @@
+# Current Limitations and Installation Alternatives
+
+At least until [#3490](https://github.com/rectorphp/rector/issues/3490) is resolved, Rector relies on _runtime PHP reflection_ to analyze your source code. That means the code Rector works on cannot merely be treated as _data_, but has to be loaded (through the autoloader) into memory to be analyzed by the PHP interpreter.
+
+> Read the [PHPStan 0.12.26 release announcement](https://phpstan.org/blog/zero-config-analysis-with-static-reflection) to learn more about runtime and static reflection. PHPStan is the static analysis tool used by Rector.
+
+This brings a few limitations and affects your choices of how you can run Rector.
+
+## PHP Language Level
+
+The PHP process that runs Rector code also needs to load and parse your source code. So, you need to run a version of PHP that is both compatible with your project code as well as Rector.
+
+This means you cannot run Rector on PHP 8 to work on a project that uses language features limited to previous PHP versions.
+
+> One example for this is [array and string offset access using curly braces](https://www.php.net/manual/de/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace).
+
+But also the other way round, you cannot run Rector on PHP 7 to analyze a project with PHP 8 language features, like [constructor property promotion](https://stitcher.io/blog/constructor-promotion-in-php-8).
+
+Together with Rector's minimum PHP 7.3 version requirement, this might limit your options of how to run Rector.
+
+Note that using an older version of Rector might allow you to run a version of PHP <7.3.
```

Comment:
That cannot be recommended, as that is older version with many missing features and invalid config loading. That could lead to random report on already fixed bugs.

Docker with PHP 7.1 downgraded and prefixed Rector should be single solution for most cases.

Reply:
`composer require --dev rector/rector` is the recommended, primary approach and the only one written directly in the README. This will – without further notice or warning – install an older version of Rector if that meets your requirements. 

For example, with PHP 7.2, you'll just get `0.8.56`.

So, if you really want to encourage people not to work with older versions (rather give Docker a try), we should state that more clearly.

By the way, what is "PHP 7.1 downgraded"? I just found https://github.com/rectorphp/rector-php71, but as far as I can see, its mentioned nowhere in the README or `docs`.

Reply:
That is current state, yes. But that's wrong. Our goal is that this command:

```php
composer require --dev rector/rector
```

Installs 0.9.x even for PHP 7.1 and also for version conflicts.
The `rector-prefixed` and `rector-php71` won't exist as they do now, but only as subtag of 0.9.x-y